Modules/posixmodule.c

```diff
@@ -2829,6 +2938,38 @@ posix_do_stat(PyObject *module, const char *function_name, path_t *path,
         fd_and_follow_symlinks_invalid("stat", path->fd, follow_symlinks))
         return NULL;
 
+#ifdef HAVE_STATX
+    struct statx stx = {};
+    static int statx_works = -1;
+    if (statx != NULL && statx_works != 0) {
+        unsigned int mask = STATX_BASIC_STATS | STATX_BTIME;
+        int flags = AT_NO_AUTOMOUNT;
+        flags |= follow_symlinks ? 0 : AT_SYMLINK_NOFOLLOW;
+
+        Py_BEGIN_ALLOW_THREADS
```

Comment:
nit: I don't see a reason not to start allowing threads a bit earlier; aren't touching the interpreter in the lines just before this